### PR TITLE
{Network} `az network nic ip-config`: Migrate commands to codegen v2

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network nic ip-config

**Description**

Total: 9 commands

**Testing Guide**

**History Notes**

[Network] BREAKING CHANGE: `az network nic ip-config`: Rename output property `privateIpAddress` to `privateIPAddress` to keep consistent with the name in API
[Network] BREAKING CHANGE: `az network nic ip-config`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
...
+        args_schema.load_balancer_inbound_nat_rules._registered = False
+        args_schema.private_ip_allocation_method._registered = False
+        args_schema.pip_obj._registered = False
+        args_schema.asgs_obj._registered = False
+        args_schema.subnet_obj._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        args.private_ip_address_version = args.private_ip_address_version
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4489]: Why assign private_ip_address_version  to itself?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
...
+        args.private_ip_allocation_method = "Static" if has_value(args.private_ip_address) else "Dynamic"
+        args.application_gateway_backend_address_pools = [{"id": x} for x in args.app_gateway_address_pools] if has_value(args.app_gateway_address_pools) else None
+        args.load_balancer_backend_address_pools = [{"id": x} for x in args.lb_address_pools] if has_value(args.lb_address_pools) else None
+        args.load_balancer_inbound_nat_rules = [{"id": x} for x in args.lb_inbound_nat_rules] if has_value(args.lb_inbound_nat_rules) else None
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4493]: Why not use `assign_aaz_list_arg`?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -4315,7 +4318,7 @@ def pre_operations(self):
         ip_configuration = {
             "name": "ipconfig1",
             "private_ip_address": args.private_ip_address,
-            "private_ip_address_version": args.private_ip_address_version if has_value(args.private_ip_address) else None,
+            "private_ip_address_version": args.private_ip_address_version,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4321]: You can add comments in the code

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 4]
```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
...
+        args_schema.private_ip_allocation_method._registered = False
+        args_schema.pip_obj._registered = False
+        args_schema.asgs_obj._registered = False
+        args_schema.subnet_obj._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.private_ip_address):
+            if args.private_ip_address == "":
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4644]: can user pass `null` value for `args.private_ip_address`?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 5]
```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
...
+        args = self.ctx.args
+        instance = self.ctx.vars.instance
+        if args.private_ip_address_version.to_serialized_data().lower() == "ipv4" and not has_value(args.subnet):
+            primary = next(x for x in instance.properties.ip_configurations if x.properties.primary)
+            args.subnet = primary.properties.subnet.id
+        if args.make_primary.to_serialized_data():
+            for config in instance.properties.ip_configurations:
+                config.properties.primary = False
+
+        args.subnet_obj = {"id": args.subnet} if has_value(args.subnet) else None
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4508]: Why use `args.subnet` as proxy for `args.subnet_obj`? Why not directly use `args.subnet_obj`?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 6]
```diff
@@ -4509,67 +4390,413 @@ def _get_nic_ip_config(nic, name):
...
+
+    def pre_operations(self):
+        args = self.ctx.args
+        args.private_ip_address_version = args.private_ip_address_version
+        args.private_ip_allocation_method = "Static" if has_value(args.private_ip_address) else "Dynamic"
+        args.application_gateway_backend_address_pools = [{"id": x} for x in args.app_gateway_address_pools] if has_value(args.app_gateway_address_pools) else None
+        args.load_balancer_backend_address_pools = [{"id": x} for x in args.lb_address_pools] if has_value(args.lb_address_pools) else None
+        args.load_balancer_inbound_nat_rules = [{"id": x} for x in args.lb_inbound_nat_rules] if has_value(args.lb_inbound_nat_rules) else None
+
+        args.pip_obj = {"id": args.public_ip_address} if has_value(args.public_ip_address) else None
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4495]: Why use `args.public_ip_address` as proxy for `args.pip_obj`?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**